app/graphql/mutations/users/update_user_details_mutation.rb

```diff
@@ -0,0 +1,29 @@
+module Mutations
+  module Users
+    class UpdateUserDetailsMutation < Mutations::BaseMutation
+      description 'Updates a User with the access_token'
+      null true
+      argument :personalAccessToken, String, required: false
+      argument :githubUsername, String, required: false
+      argument :organizationId, Int, required: false
+      field :user, Types::UserType, null: true
+      field :errors, [String], null: true
+      def resolve(data)
+        current_user = context[:current_user]
+        if current_user
+          begin
+            current_user.update!(data.slice(:personal_access_token, :github_username, :organization_id))
+          rescue ActiveRecord::RecordInvalid => e
+            return user_detail_response(nil, e.record.errors.full_messages)
+          end
+          user_detail_response(current_user)
+        end
+        user_detail_response
```

Comment:
When there is a current user present, it will still return this, instead of including the current user's updated details.  And actually, the `rescue` can be defined by the method's `def` and `end`, instead of the `begin/end`.

Here is one way you could refactor this, if you like:

```ruby
      def resolve(data)
        current_user = context[:current_user]
        attributes   = data.slice(:personal_access_token, :github_username, :organization_id)

        return user_detail_response(nil, ['Missing current user']) if current_user.blank?

        current_user.update!(attributes)
        user_detail_response(current_user)
      rescue ActiveRecord::RecordInvalid => e
        user_detail_response(nil, e.record.errors.full_messages)
      end
```

Reply:
ha really need to return a 401, but not going to get into how to do that in graphql lol

Reply:
updated and cleaned up